sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcRelationProvider.scala

```diff
@@ -82,7 +81,7 @@ class JdbcRelationProvider extends CreatableRelationProvider
 
       val (doCreate, doSave) = (mode, tableExists) match {
         case (SaveMode.Ignore, true) => (false, false)
-        case (SaveMode.ErrorIfExists, true) => throw new TableAlreadyExistsException(
+        case (SaveMode.ErrorIfExists, true) => throw new SQLException(
           s"Table $table already exists, and SaveMode is set to ErrorIfExists.")
```

Comment:
Normally, we issue an `AnalysisException`. `Table or view $table already exists`